observer.ts

```diff
@@ -127,37 +127,47 @@ export async function observer(
     const topicsBytes = log.topics.map(toBytes).map(Buffer.from);
     const [abiHash, topic1, topic2, topic3] = topicsBytes;
 
-    await prisma.event.create({
-      data: {
-        blockTimestamp: new Date(Number(timestamp) * 1000),
-        txIndex: log.transactionIndex,
-        logIndex: log.logIndex,
-        blockNumber: Number(log.blockNumber),
-        blockHash: Buffer.from(blockHashBytes),
-        txHash: Buffer.from(toBytes(log.transactionHash)),
-        sourceAddress: Buffer.from(addressBytes),
-        abiHash,
-        topic1,
-        topic2,
-        topic3,
-        data: Buffer.from(toBytes(log.data)),
-      },
-    });
+    try {
+      await prisma.event.create({
+        data: {
+          blockTimestamp: new Date(Number(timestamp) * 1000),
+          txIndex: log.transactionIndex,
+          logIndex: log.logIndex,
+          blockNumber: Number(log.blockNumber),
+          blockHash: Buffer.from(blockHashBytes),
+          txHash: Buffer.from(toBytes(log.transactionHash)),
+          sourceAddress: Buffer.from(addressBytes),
+          abiHash,
+          topic1,
+          topic2,
+          topic3,
+          data: Buffer.from(toBytes(log.data)),
+        },
+      });
 
-    amqpChannel.publish(
-      { routingKey: EvmEventsQueueName },
-      { contentType: "application/octet-stream" },
-      serializeEventMessage({
-        address: addressBytes,
-        sigHash: abiHash,
-        topics: topicsBytes,
-        blockTimestamp: timestamp,
-        txIndex: BigInt(log.transactionIndex),
-        logIndex: BigInt(log.logIndex),
-        blockNumber: log.blockNumber,
-        blockHash: blockHashBytes,
-      }),
-    );
+      amqpChannel.publish(
+        { routingKey: EvmEventsQueueName },
+        { contentType: "application/octet-stream" },
+        serializeEventMessage({
+          address: addressBytes,
+          sigHash: abiHash,
+          topics: topicsBytes,
+          blockTimestamp: timestamp,
+          txIndex: BigInt(log.transactionIndex),
+          logIndex: BigInt(log.logIndex),
+          blockNumber: log.blockNumber,
+          blockHash: blockHashBytes,
+        }),
+      );
+    } catch (e) {
+      if (
```

Comment:
Is this check still needed? if so, we might also need to leave a comment about the reason for future readers.

Reply:
And, Manual logging for consumed exceptions would be needed. (for debugging situations)

Reply:
We're checking to see if there's any other exceptions missed, that might happen in some way. Regarding the log, yes I will keep in mind 😄